Cython/Compiler/ExprNodes.py

```diff
@@ -2899,12 +2899,30 @@ def analyse_types(self, env):
         if self.expr_scope:
             env = self.expr_scope  # actually evaluate sequence in this scope instead
         self.sequence = self.sequence.analyse_types(env)
+        def have_reversed_cpp_iteration():
+            """
+            Returns True if it represents a reversed iteration over a C++ sequence.
+
+            This supports C++ classes with reversed_iterator implemented.
+            """
+            reversed_cpp_iterable = False
+            if isinstance(self.sequence, SimpleCallNode) and self.sequence.function.name == "reversed":
```

Comment:
I think you need an extra check here. Not all `SimpleCallNode` hold names (some may be attributes for example)